application/models/SurveyDynamic.php

```diff
@@ -907,19 +917,19 @@ public function getQuestionArray($oQuestion, $oResponses, $bHonorConditions, $su
         }
 
 
-        if ($oQuestion->parent_qid != 0 && $oQuestion->parents['type'] === "1") {
+        if ($oQuestion->parent_qid != 0 && $oQuestion->parent['type'] === "1") {
 
             $aAnswers = (
                 $oQuestion->parent_qid == 0
                     ? $oQuestion->answers
-                    : ($oQuestion->parents != null
-                        ? $oQuestion->parents->answers
+                    : ($oQuestion->parent != null
+                        ? $oQuestion->parent->answers
                         : []
                     )
             );
 
             foreach($aAnswers as $key=>$value){
-                $aAnswerText[$value['code']] = $value['answer'];
+                $aAnswerText[$value['code']] = Answer::model()->getAnswerFromCode($value->qid, $value->code, $sLanguage, $value->scale_id);
```

Comment:
Is this database query needed here, if we already did `$oQuestion->answers` above?